src/components/inputs/ui-select-extended/ui-select-extended.component.tsx

```diff
@@ -85,10 +102,22 @@ const UiSelectExtended: React.FC<FormFieldProps> = ({ question, handler, onChang
     // If not searchable, preload the items
     if (dataSource && !isTrue(question.questionOptions.isSearchable)) {
       setIsLoading(true);
-      dataSource.fetchData(null, config).then((dataItems) => {
-        setItems(dataItems.map(dataSource.toUuidAndDisplay));
-        setIsLoading(false);
-      });
+      if (question.type === 'programState') {
+        dataSource.fetchData(workFlowMeta, encounterContext.programUuid).then((dataItems) => {
+          setItems(dataItems.map(dataSource.toUuidAndDisplay));
+          setIsLoading(false);
+        });
+      } else if (question.type === 'programWorkflow') {
+        dataSource.fetchData('', encounterContext.programUuid).then((dataItems) => {
+          setItems(dataItems.map(dataSource.toUuidAndDisplay));
+          setIsLoading(false);
+        });
+      } else {
```

Comment:
This logic is making the ui-select component aware of the program workflows which introduces some kind of coupling. Don't the data sources suffice?

Reply:
The `programWorkflow` field uuid needs to be passed to the `programState ` datasource everytime a user selects a new program workflow, thats why the logic exisits here. The fields are dependent on each other.

Reply:
Why not simply pass the config to the datasource and let the datasource to apply the necessary conditions? Additionally, I think the datasource will need to the question object, we can add the question object to the config or revise the Datasource interface to optionally accept a question object.